### DISCUSSION TITLE
커뮤니케이션 템플릿 픽스쳐 생성 시 communication_template.json 사용 논의

### DISCUSSION THREAD

**Discussion by yeseul-cho** [8percent/styleguide#49] [General]
테스트에서 커뮤니케이션 템플릿이 필요한 경우, 두 가지 방법으로 픽스쳐를 만들고 있습니다.
1. communication_template.json 사용
2. 테스트코드에 커뮤니케이션 템플릿 직접 작성

아주 옛날에 작성된 코드는 1번, 최근에 작성된 코드는 주로 2번을 사용하고 있습니다. 
두 가지 방법 중 하나를 정해서 점차 통일해나가면 좋을 것 같은데 의견 부탁드립니다. 

[1번 장점]  
- 템플릿을 일일이 테스트에 작성하는 수고를 덜 수 있음. 
- 실제 프로그램 실행 환경과 비슷함(DB에 있는 커뮤니케이션 템플릿 내용을 서버에서 알 수 없다는 점에서)
- 여러 개의 템플릿 업데이트 시, json 파일 업데이트가 쉬움

[1번 단점]
- 한 개의 템플릿만 필요한 테스트에도 150개 가량의 테스트가 전부 생성됨(매번 생성하지 않도록 수정도 가능)
- 어느 테스트에 어느 템플릿이 사용되는 지 모호하게 됨 => 가독성 저해

[2번 장점]
- 각 테스트를 독립적으로 정의할 수 있음.
- 어느 테스트에 어느 템플릿이 사용하는지 명시할 수 있음.

[2번 단점]
- 템플릿 내용을 테스트에 일일이 작성해야 함
- 템플릿 업데이트 시 일일이 수정해야 함

**Comment by Kirade** [comment]
저는 2번 방식을 조금 더 선호합니다.

1번 방식에서 사용하는 communication_template.json 파일은 명시적이지 않아 잘 관리되기 어려운 부분이 있기 때문입니다.
여느 다른 테스트들을 작성하는 방식과 비슷하게  2번 방식으로 테스트 케이스에서 명시적으로 알 수 있도록 작성하는 편이 좋을것 같습니다.

2번 방식을 사용하되 1번 방식의 장점을 잘 채용할 수 있는 방식도 있을것 같은데요.
공통적으로 사용될 수 있는 여러 커뮤니케이션 템플릿을 테스트 환경 내의 공용 Fixture 로 정의해서 필요한 부분에서 사용하게 된다면 2번 방식의 단점인 테스트에 매번 같은 내용을 작성하거나 변경하는 수고를 덜 수 있을것 같습니다.
